File: services/events/app/schemas/assistant.py
from pydantic import BaseModel, Field


class AssistantChatMessage(BaseModel):
    role: str = Field(pattern="^(user|assistant)$")
    content: str = Field(min_length=1, max_length=4000)


class AssistantChatRequest(BaseModel):
    message: str = Field(min_length=1, max_length=4000)
    history: list[AssistantChatMessage] = Field(default_factory=list)


class AssistantContextSummary(BaseModel):
    events_total: int = 0
    search_events: int = 0


class AssistantChatResponse(BaseModel):
    reply: str
    source: str
    model: str | None = None
    context_summary: AssistantContextSummary | None = None

File: frontend/src/api/assistant.ts
import { apiFetch } from "./client";

export interface AssistantChatMessage {
  role: "user" | "assistant";
  content: string;
}

export interface AssistantChatResponse {
  reply: string;
  source: "nvidia" | "local" | string;
  model: string | null;
  context_summary?: {
    events_total: number;
    search_events: number;
  };
}

export function assistantChat(
  token: string | null,
  message: string,
  history: AssistantChatMessage[] = [],
) {
  return apiFetch<AssistantChatResponse>("/v1/assistant/chat", token, {
    method: "POST",
    body: JSON.stringify({ message, history }),
  });
}
